### PR TITLE
Make xargs chainable

### LINKED ISSUE #7: Make xargs chainable
https://github.com/impactaky/dax_extras/pull/6#issuecomment-1606270396

### PR DESCRIPTION
**Pull request by sigmaSd**
fix https://github.com/impactaky/dax_extras/issues/7

### CODE REVIEW

**Comment by impactaky** [comment]
Does it work well when multiple lines are given to xargs?

**Comment by sigmaSd** [comment]
the main branch doens't seem to handle multiline 

what I expect is the same as xargs in linux so
```sh
echo "hello
                                     hello2" | xargs -i echo {} world
 ```
 should give
 ```
 hello world
hello2 world
```

but in main brach:
```ts
  const resultMultiLine = await $`echo "hello\nhello2"`
    .xargs((input) => $`echo ${input} world`.stdout("piped"));

  console.log(resultMultiLine[0].stdout);
```
result
```
hello world
```
it seems like only the first line is outputed



and with this pr
```ts
  const resultMultiLine = await $`echo "hello\nhello2"`
    .xargs((input) => $`echo ${input} world`.stdout("piped"));

  console.log(await resultMultiLine.result.text());
```
result
```
hello2 world
```
so it seems to keeping only the last line


Maybe this issue is in LineStream and is unrelated to this pr

**Comment by impactaky** [comment]
This PR destroys the following cases.
`echo "line1\nline2" | xargs -I echo {}`
I should write these cases in the test, but currently there is a lack of
functionality for writing tests easily, so we will give priority to that.

**Comment by sigmaSd** [comment]
>echo "line1\nline2" | xargs -I echo {}

but this case already doesn't seem to work correctly in main branch
```ts
    const result = await $`echo "line1\nline2"`
      .xargs((input) => $`echo ${input} world`.stdout("piped"));
    console.log(result[0].stdout);
```
it outputs 
```
line1 world
```
instead of the expected
```
line1 world
line2 world
```

**Comment by impactaky** [comment]
You can see `line2 world` on result[1].stdout.

**Comment by sigmaSd** [comment]
Interesting , kind of weird to get the result spited on separate CommandResult

**Comment by sigmaSd** [comment]
I update the pr to match the main branch behavior, now multi line should work the same

Hopefully we can improve that since its still weird the lines are split on the array

**Comment by sigmaSd** [comment]
But now I look more at the code, I can see that you seem to actually want multi-line to work like this, so maybe its fine

File: command_builder.test.ts [hunk 1]
```diff
@@ -10,7 +10,16 @@ Deno.test("Quick example", async () => {
...
+      .xargs((input) => $`echo ${input} world`)
+      .then((output) => output[0].xargs((input) => $`echo ${input} world2`));
+    assertEquals(await result[0].text(), "hello world world2");
+  }
+  {
+    const result = await $`echo "line1\nline2"`
+      .xargs((input) => $`echo ${input} world`);
+    assertEquals(await result[0].text(), "line1 world");
+    assertEquals(await result[1].text(), "line2 world");
+  }
```
Comment (impactaky) [inline command_builder.test.ts]: Thank you for adding test!
Can you separate each block into a separate Deno.test?

Comment (sigmaSd) [reply command_builder.test.ts]: done

**Comment by impactaky** [comment]
Thank you!

**Pull request merged**